sklearn/neural_network/tests/test_base.py

```diff
@@ -26,5 +27,52 @@ def test_binary_log_loss_1_prob_finite():
 )
 def test_log_loss_1_prob_finite(y_true, y_prob):
     # y_proba is equal to 1 should result in a finite logloss
-    loss = log_loss(y_true, y_prob)
+    loss = log_loss(y_true, y_prob, sample_weight=np.ones(shape=y_prob.shape[0]))
     assert np.isfinite(loss)
+
+
+def test_loss_functions():
```

Comment:
I'll prefer separate test for each loss function and test cases where 

- sample weight is zero, i.e. `sample_weight=[0, 1, 1, 0]`
- sample weight is different, i.e. `sample_weight=[1, 2, 3]`